src/material/table/testing/table-harness.ts

```diff
@@ -0,0 +1,100 @@
+/**
+ * @license
+ * Copyright Google LLC All Rights Reserved.
+ *
+ * Use of this source code is governed by an MIT-style license that can be
+ * found in the LICENSE file at https://angular.io/license
+ */
+
+import {ComponentHarness, HarnessPredicate} from '@angular/cdk/testing';
+import {TableHarnessFilters, RowHarnessFilters} from './table-harness-filters';
+import {MatRowHarness, MatHeaderRowHarness, MatFooterRowHarness} from './row-harness';
+
+/** Data extracted from a table organized by columns. */
+export interface MatTableHarnessColumnsText {
+  [columnName: string]: {
+    text: string[];
+    headerText?: string;
```

Comment:
continued from https://github.com/angular/components/pull/17799#discussion_r358430226:

It feels wrong to me to just grab the first one, that's not what I would expect if I had a table with multiple header or footer rows. @jelbourn @andrewseguin do you have an opinion on this one?

Reply:
Changed to be a string array.